lib/src/network/coap_network_tinydtls.dart

```diff
@@ -166,4 +157,14 @@ class CoapNetworkTinyDtls implements CoapINetwork {
   void close() {
     _dtlsClient?.close();
   }
+
+  Future<void> rebind() async {
+    try {
+      close();
+      await bind();
+    } on Exception catch (_) {
+      // Fail silently (try again on next request)
+    }
+    _eventBus.fire(CoapSocketRebindEvent());
+  }
```

Comment:
I think we might need a broader refactor, but that might be better saved for another PR. For example, I don't think you just `bind` for TCP (since it's not a connectionless protocol you `connect`), so when we support TCP this method might be a bit out-of-place on the top-level. I would use more generic names at the top level, and more specific names below.

A DTLS socket is also a UDP socket at the end of the day, I would rename `CoapNetworkOpenSSL` and perhaps also `CoapNetworkTinyDtls` (although implied since DTLS is UDP) to indicate that they are UDP networks/sockets, and have them extend `CoapNetworkUDP`.

This way, `CoapNetworkUDP` manages all raw UDP sockets, and this socket is passed into the DTLS clients when they are created (which is also possible with tinydtls). What do you think?

Reply:
Very good points! Sounds like a good approach to me :)